### PR TITLE
support scoreExtensions

### PR DESCRIPTION
**Pull request by Gekko0114**
#### What type of PR is this?

/kind feature
/kind api-change

#### What this PR does / why we need it:
Support score Extensions

#### Which issue(s) this PR fixes:
Part of #72 

#### Does this PR introduce a user-facing change?
```release-note
support score Extensions
```

#### What are the benchmark results of this change?

```benchstat

```

### CODE REVIEW

**Comment by Gekko0114** [comment]
Hi @sanposhiho,

I've created a PR for scoreExtensions. PTAL, though I haven't implemented cache of each node's score.

File: guest/api/types.go [hunk 1]
```diff
@@ -91,9 +91,20 @@ type ScorePlugin interface {
...
+
 type NodeInfo interface {
 	// Metadata is a convenience that triggers Get.
 	proto.Metadata
 
 	Node() proto.Node
 }
+
+type NodeScoreList interface {
+	NodeScoreList() map[string]int
```
Comment (sanposhiho) [inline guest/api/types.go:109]: It's also a similar one, let's change like:

```suggestion
type NodeScore interface {
	// Map returns a map
	// which is keyed by the node name and valued by the score.
	Map() map[string]int
```

Comment (sanposhiho) [reply guest/api/types.go:109]: It's a general suggestion, when you use map in interface, it's always better to say explicitly what is key and what is value.

File: guest/scoreextensions/scoreextensions.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+//	}
+//
+//	type scoreExtensionsPlugin struct{}
+//
+//	func (scoreExtensionsPlugin) NormalizeScore(state api.CycleState, pod api.Pod, nodeScoreList map[string]int) (map[string]int, *api.Status) {
+//		panic("implement me")
+//	}
+func SetPlugin(scoreExtensions api.ScoreExtensions) {
+	if scoreExtensions == nil {
+		panic("nil scorePlugin")
```
Comment (sanposhiho) [inline guest/scoreextensions/scoreextensions.go:51]:
```suggestion
		panic("nil scoreExtensions")
```

**Review by sanposhiho — COMMENTED** [review]
/lgtm

One nit picked, lgtm otherwise

File: guest/api/types.go (hunk 1 above)
Comment (Gekko0114) [reply guest/api/types.go]: Thanks, fixed it

**Comment by Gekko0114** [comment]
Hi @sanposhiho ,
Thanks for your review. Fixed it, PTAL

**Pull request merged**